apps/s_client.c

```diff
@@ -2047,14 +2058,15 @@ int s_client_main(int argc, char **argv)
 #endif
             sbio = BIO_new_dgram(sock, BIO_NOCLOSE);
 
-        if ((peer_info.addr = BIO_ADDR_new()) == NULL) {
+        if (sbio == NULL || (peer_info.addr = BIO_ADDR_new()) == NULL) {
             BIO_printf(bio_err, "memory allocation failure\n");
```

Comment:
if `sbio` is non NULL but the `BIO_ADDR_new()` call fails then this will leak `sbio`